### ISSUE TITLE
Suggest you use `Async.run` here, or allow user to pass task as keyword argument.

### ISSUE THREAD

**Issue by ioquatix** [matti/async-nats#1]
https://github.com/matti/async-nats/blob/a6dd9b4c8b11e4242fbbb0713dd490d034f0658d/lib/async/nats/client.rb#L153

I see that I should probably document the preferred patterns to use. Otherwise it's tricky for users to reason about the code.

One way to minimise this is to use the async-await gem which provides some built in high level patterns.

https://github.com/socketry/async-await/blob/master/README.md

**Comment by matti** [comment]
Ok, but _why_ is async-await better than this?

**Comment by ioquatix** [comment]
Because it forces you to use only a single abstraction rather than inventing your own.

**Comment by matti** [comment]
yeah so this is definitely a documentation issue - I have no idea which patterns to use. I came up with this by reading the async source...

Ok, I give this a try

**Comment by ioquatix** [comment]
Don't take this the wrong way, I love that you are trying things and exploring the problem domain.

I just think it's important that gems in the async namespace follow certain patterns, otherwise users are going to have trouble making assumptions about how they work.

**Comment by matti** [comment]
and to be more specific: I found out async-nats, but the README has some random example with chickens.

**Comment by ioquatix** [comment]
I've been working on the documentation: https://github.com/socketry/async#usage

**Comment by matti** [comment]
@ioquatix nice! that is helpful. I've been doing other projects with async-* - I'll revisit this nats gem later.
